phoenix-core/src/main/java/org/apache/phoenix/compile/ScanRanges.java

```diff
@@ -385,9 +387,25 @@ public Scan intersectScan(Scan scan, final byte[] originalStartKey, final byte[]
         if (scanStopKey.length > 0 && Bytes.compareTo(scanStartKey, scanStopKey) >= 0) { 
             return null; 
         }
-        newScan.setAttribute(SCAN_ACTUAL_START_ROW, scanStartKey);
-        newScan.setStartRow(scanStartKey);
-        newScan.setStopRow(scanStopKey);
+        if(ScanUtil.isLocalIndex(scan)) {
```

Comment:
The code in this if statement can go in the if (keyOffset > 0) if statement instead. We're trying to avoid checks for local indexes.